src/MPAGSCipher/ProcessCommandLine.cpp

```diff
@@ -1,13 +1,14 @@
 #include "ProcessCommandLine.hpp"
+#include "CipherMode.hpp"
 
 #include <iostream>
 #include <string>
 #include <vector>
 
+
+
 bool processCommandLine(const std::vector<std::string>& cmdLineArgs,
-                        bool& helpRequested, bool& versionRequested,
-                        std::string& inputFile, std::string& outputFile,
-                        std::string& cipherKey, bool& encrypt)
+                        ProgramSettings& CmdSettings)
```

Comment:
According to our naming convention, `CmdSettings` should be in lowerCamelCase because it's a variable name:
```suggestion
                        ProgramSettings& cmdSettings)
```